src/img2pdf.py

```
@@ -347,21 +376,80 @@ def valid_date(string):
+        "b1"          : "2004x2004^",   # "2004x2836",
+        "b2"          : "1418x1418^",   # "1418x2004",
+        "b3"          : "1002x1002^",   # "1002x1418",
+        "b4"          : "709x709^",     # "709x1002",
+        "b5"          : "501x501^",     # "501x709",
+        "c0"          : "2600x2600^",   # "2600x3677",
+        "c1"          : "1837x1837^",   # "1837x2600",
+        "c2"          : "1298x1298^",   # "1298x1837",
+        "c3"          : "918x918^",     # "918x1298",
+        "c4"          : "649x649^",     # "649x918",
+        "c5"          : "459x459^",     # "459x649",
+        "c6"          : "323x323^",     # "323x459",
+        "flsa"        : "612x612^",     # "612x936",
+        "flse"        : "612x612^",     # "612x936",
+        "halfletter"  : "396x396^",     # "396x612",
+        "tabloid"     : "792x792^",     # "792x1224",
+        "statement"   : "396x396^",     # "396x612",
+        "executive"   : "540x540^",     # "540x720",
+        "folio"       : "612x612^",     # "612x936",
+        "quarto"      : "610x610^",     # "610x780"
+    }
+
+    try:
+        string = papersizes[string.lower()]
+    except KeyError, e:
+        pass
+
+    pattern = re.compile('([0-9]*\.?[0-9]*)([a-z]*)x([0-9]*\.?[0-9]*)([a-z]*)([^0-9a-z]*)')
+    tokens = pattern.match(string)
+
+    # tokens.group(0) == should match entire input string
+    # tokens.group(1) == x (width)
+    # tokens.group(2) == units -- not currently used; defaults to points
+    # tokens.group(3) == y (height)
+    # tokens.group(4) == units -- not currently used; defaults to points
+    # tokens.group(5) == extra options
+
+    if tokens.group(0) == '':
         msg = "input size needs to be of the format Ax, xB or AxB with A and B being integers"
         raise argparse.ArgumentTypeError(msg)
-    x = tokens[0]
-    y = tokens[1]
-    if x == '':
+
+    try:
+        x = float(tokens.group(1))
+    except ValueError, e:
         x = None
```

If something that matches the regex `[0-9]*\.?[0-9]*` cannot be parsed as float, then something went seriously wrong and you should throw an error accordingly.